app/docker/helpers/constraintsHelper.js

```diff
@@ -0,0 +1,107 @@
+function ConstraintModel(op, key, value) {
+  this.op = op;
+  this.value = value;
+  this.key = key;
+}
+
+var patterns = {
+  id: {
+    nodeId: 'node.id',
+    nodeHostname: 'node.hostname',
+    nodeRole: 'node.role',
+    nodeLabels: 'node.labels.',
+    engineLabels: 'engine.labels.'
+  },
+  op: {
+    eq: '==',
+    neq: '!='
+  }
+};
+
+function matchesConstraint(value, constraint) {
+  if (constraint && ((constraint.op === patterns.op.eq &&
+        value !== constraint.value) ||
+      (constraint.op === patterns.op.neq &&
```

Comment:
Conditions that are part of the same check (inside the same parenthesis group) should be on the same line, easier to read.